vector/src/main/java/im/vector/app/core/pushers/UnifiedPushHelper.kt

```diff
@@ -92,16 +92,21 @@ class UnifiedPushHelper @Inject constructor(
                 return@launch
             }
 
-            // By default, use internal solution (fcm/background sync)
-            UnifiedPush.saveDistributor(context, context.packageName)
             val distributors = UnifiedPush.getDistributors(context)
 
             if (distributors.size == 1 && !force) {
                 UnifiedPush.saveDistributor(context, distributors.first())
                 UnifiedPush.registerApp(context)
                 onDoneRunnable?.run()
             } else {
-                openDistributorDialogInternal(activity, pushersManager, onDoneRunnable, distributors, !force, !force)
+                openDistributorDialogInternal(
+                        activity = activity,
+                        pushersManager = pushersManager,
+                        onDoneRunnable = onDoneRunnable,
+                        distributors = distributors,
+                        unregisterFirst = force,
```

Comment:
Just to highlight that `unregisterFirst` parameter has been changed from `!force` to `force`.

Reply:
correct, it's intentional, the unified push connector crashes when attempting to remove a "null" token

Reply:
@ouchadam : I've seen your comment and I can't find a way to crash it with a null token. Anything relevant to fix your issue in the lib is welcome

Reply:
using the parent commit for this PR a251002808921a1da44556795b073a7449257c27 and commenting out the default `UnifiedPush.saveDistributor(context, context.packageName)` https://github.com/vector-im/element-android/pull/6890/commits/0a2ce698becd39abddc85ca4263ea57a211ca052#diff-d0af9539e5621c00bfa7f4315721ab35c43af91f06e81d8f8a8b097d51a45487L96 

when unregistering during the dialog flow, this crash happens

```
Thread: main, Exception: java.lang.RuntimeException: Unable to start receiver im.vector.app.push.fcm.EmbeddedFCMDistributor: java.lang.NullPointerException
        at android.app.ActivityThread.handleReceiver(ActivityThread.java:3855)
        at android.app.ActivityThread.access$1500(ActivityThread.java:224)
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1898)
        at android.os.Handler.dispatchMessage(Handler.java:107)
        at android.os.Looper.loop(Looper.java:224)
        at android.app.ActivityThread.main(ActivityThread.java:7562)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:539)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:950)
     Caused by: java.lang.NullPointerException
        at org.unifiedpush.android.embedded_fcm_distributor.EmbeddedDistributorReceiver.onReceive(EmbeddedDistributorReceiver.kt:22)
        at android.app.ActivityThread.handleReceiver(ActivityThread.java:3846)
        at android.app.ActivityThread.access$1500(ActivityThread.java:224) 
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1898) 
        at android.os.Handler.dispatchMessage(Handler.java:107) 
        at android.os.Looper.loop(Looper.java:224) 
        at android.app.ActivityThread.main(ActivityThread.java:7562) 
        at java.lang.reflect.Method.invoke(Native Method) 
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:539) 
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:950) 
```

| GIF | 
| --- |
![up-remove-crash](https://user-images.githubusercontent.com/1848238/186618466-549cffe4-1bc8-4ee0-bd44-e73e87a3dd6e.gif)

Reply:
Thanks, I've spotted the issue, in the embedded distributor. I'm fixing it tonight

Reply:
great to hear, thanks! :raised_hands:

Reply:
done with android-embedded_fcm_distributor:2.1.2